Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp

```diff
@@ -245,28 +233,33 @@ void ProcessLauncher::launchProcess()
         g_error("Unable to spawn a new child process: %s", error->message);
 
 #if OS(LINUX)
-    GRefPtr<GSocket> pidSocket = adoptGRef(g_socket_new_from_fd(pidSocketPair.server.release(), &error.outPtr()));
-    if (!pidSocket)
-        // Note: g_socket_new_from_fd() takes ownership of the fd only on success, so if this error
-        // were not fatal, we would need to close it here.
-        g_error("Failed to create pid socket wrapper: %s", error->message);
-
-    // We need to get the pid of the actual WebKit auxiliary process, not the bwrap or flatpak-spawn
-    // intermediate process. And do it without blocking, because process launching is slow.
-    g_socket_set_blocking(pidSocket.get(), FALSE);
-    m_socketMonitor.start(pidSocket.get(), G_IO_IN, RunLoop::mainSingleton(), [protectedThis = Ref { *this }, this, pidSocket, serverSocket = WTFMove(webkitSocketPair.server)](GIOCondition condition) mutable -> gboolean {
-        if (!(condition & G_IO_IN))
-            g_error("Failed to read pid from child process");
-
-        m_processID = IPC::readPIDFromPeer(g_socket_get_fd(pidSocket.get()));
-        RELEASE_ASSERT(m_processID);
-
-        m_socketMonitor.stop();
+    if (isInsideFlatpak() || (sandboxEnabled && shouldUseBubblewrap())) {
+        GRefPtr<GSocket> pidSocket = adoptGRef(g_socket_new_from_fd(webkitSocketPair.server.release(), &error.outPtr()));
```

Comment:
Previously we used a separate pid socket here, but now it's just the normal connection socket, so this name doesn't make sense anymore. Maybe you could call it just `socket` instead of `pidSocket`.